task_schedule.pl

```diff
@@ -196,9 +197,20 @@
     # Update heartbeat file
     eval { 	io($opt{heartbeat})->write(qq{Running on $hostname}); };
     if ($@) {
-		heartbeat_perm_fail($@);
+		email_and_die($@, "ERROR - Cannot update heartbeat");
     }
 
+    eval{
+        my $cwd = getcwd();
+        if ($cwd eq ''){
+            die "cwd is $cwd";
```

Comment:
Just change this line to `die "cwd is ''";` and I think this PR is good for merge.